### ISSUE TITLE
`populate_connections` doesn't fetch `datasource_id`

### ISSUE THREAD

**Issue by LuigiCerone** [tableau/server-client-python#1536]
**Describe the bug**

Hi,

I'm encountering an issue with the Tableau Python client. Specifically, the `datasource_id` is missing from the `ConnectionItem` object.

Here’s the scenario: after retrieving a datasource by id, I populate its connections, but the `ConnectionItem` objects lack the datasource information.

From reviewing the source code:
https://github.com/tableau/server-client-python/blob/1d98fdad189ebed130fb904e8fa5dca2207f9011/tableauserverclient/server/endpoint/datasources_endpoint.py#L106
it seems the REST method being used by the module is: `GET /api/api-version/sites/site-id/datasources/datasource-id/connections`.

According to the <a href="https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_data_sources.htm#query_data_sources" target="_blank">official documentation</a>, this method does not appear to return datasource information.

**Versions**
 - Tableau Cloud 2024.3
 - Python version: 3.11.0
 - TSC library version: 0.34

**To Reproduce**

```python
import os
import tableauserverclient as TSC

token_name = 'my_token'
token_value = os.environ['TABLEAU_TOKEN']
tableau_site = 'my-site-it'
tableau_server = 'https://eu-west-1a.online.tableau.com'
datasource_id = 'REPLACE_ME'


def main():
    tableau_auth = TSC.PersonalAccessTokenAuth(token_name, token_value, site_id=tableau_site)
    server = TSC.Server(tableau_server, use_server_version=True)
    with server.auth.sign_in(tableau_auth):
        ds = server.datasources.get_by_id(datasource_id)
        server.datasources.populate_connections(ds)

        # This will fail
        assert ds.connections[0].datasource_id


if __name__ == '__main__':
    main()
```

**Results**

`connections[0].datasource_id` is `None`, data is not fetched from the REST API.

Thanks :)

**Comment by jacalata** [comment]
Yes, that's a bug. We can populate the id from the datasource information we already have, I can add that.

**Comment by jorwoods** [comment]
@LuigiCerone what are you expecting that datasource_id to be? The id of the datasource on which you called `populate_connections`? Or an upstream datasource that the connection pulls data from?

**Comment by LuigiCerone** [comment]
Hello @jorwoods 

> The id of the datasource on which you called populate_connections?

This.

**Comment by jorwoods** [comment]
@LuigiCerone can you help me understand what this enables beyond using the id or datasource object you already have?

@jacalata I propose we remove the datasource_id and name from the connection item entirely. Theoretically a breaking change but since it's never been populated, usage should be basically non-existent. My thinking is:

1. If we're going to wire up datasource id and name, then we should also do workbook_id and name. 
2. Different names breaks symmetry
3. Having different names for the originating object means the users script would have to know the type of the originating object.
4. Different names means we have two null attributes half of the time, that cannot be populated in a way that makes any sense.

**Comment by LuigiCerone** [comment]
> @LuigiCerone can you help me understand what this enables beyond using the id or datasource object you already have?

The issue regarding the missing datasource value within the collection came up because I was trying to update the hyper data [via the API ](https://tableau.github.io/server-client-python/docs/api-ref#datasourceupdate_hyper_data) using the method:

```python
job = server.datasources.(
    ds.connections[0],
    request_id=request_id,
    actions=actions,
    payload=payload
)
```

Inside `update_hyper_data` implementation there is this:

https://github.com/tableau/server-client-python/blob/1d98fdad189ebed130fb904e8fa5dca2207f9011/tableauserverclient/server/endpoint/datasources_endpoint.py#L330-L333

which fails with the aformentioned problem.

**Comment by jorwoods** [comment]
Based on the legitimate and internal usage of the `datasource_id`, I withdraw my proposal of removing those attributes. I submitted a PR to carry through the datasource id and name.
